src/formik-persist.tsx

```diff
@@ -31,8 +38,10 @@ export class Persist extends React.Component<PersistProps, {}> {
     }
   }
 
-  componentDidMount() {
-    const maybeState = window.localStorage.getItem(this.props.name);
+  async componentDidMount() {
```

Comment:
Using async here can lead to inconsistent state. Stick with`.then(success => {}, error => {})`